### DISCUSSION TITLE
Is it possible to use Prisma with the PlanetScale Serverless Driver?

### DISCUSSION THREAD

**Discussion by kirkouimet** [planetscale/discussion#292] [Q&A] [answered]
We are using Cloudflare Works with the PlanetScale Serverless Driver:

https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript

We would like to adopt Prisma as our ORM. Prisma currently requires a direct MySQL connection, which Cloudflare workers do not support. The alternative is to use Prisma's Data Proxy, but we would like to avoid having to use this as PlanetScale's serverless driver already handles connection pooling.

Is it possible to create a custom connection in Prisma which uses PlanetScale's serverless driver?

Thanks!

**Comment by orware** [comment, marked as answer]
Hello @kirkouimet,

Something for Prisma may not be in the works immediately for our serverless driver since that would be dependent on the Prisma team's ability to develop that capability at this time to my understanding.

That being said, a Kysely dialect for the PlanetScale serverless driver has been created and shared here a few weeks ago:
https://github.com/depot/kysely-planetscale

And it might provide a middle-ground ORM solution that can be utilized for your project's needs!

**Comment by kirkouimet** [comment]
Thanks @orware! I'll reach out to the Prisma team to see how easy it would be to create a connection in Prisma that leverages the PlanetScale Serverless Driver.

**Reply by jcohenho** [reply]
@kirkouimet please post an update if you find out more! Thank you!

**Comment by Izhaki** [comment]
Prisma still do not support PlanetScale Serverless driver, although you can connect to PlanetScale from non-edge functions (normal NodeJS runtime).

Edge function support for Prisma Client does not seem likely soon, because internally the query engine is... urm... quite massive (albeit for good reasons).

https://github.com/prisma/prisma/issues/15265.

Prisma also has `@prisma/client/edge`, but it is locked to their data-proxy product, which is, if I understand correctly, very similar and hence redundant to what PlanetScale do (http connection into a mysql connection pool).

---

Due to this, we have limited the Prisma Client usage for development only:

- We use the client API to see what SQL queries are generated (just local scripts). We then take the queries produces by the client and just hit PlanetScale with its serverless driver. "Stitching" manually the results, if needed.
- We use it to generate types and import them using `import type` so the `@prisma/client` does not gets bundled in production (into edge functions).

**Reply by janpio** [reply]
That is a fair approach while Prisma does not support the PlanetScale Serverless Driver directly.
Let us @prisma know if you ever run into any limitations with this approach.
